interactions/context.py

```diff
@@ -708,57 +717,8 @@ async def func():
 
         return await func()
 
-    async def popup(self, modal: Modal) -> None:
-        """
-        This "pops up" a modal to present information back to the
-        user.
-
-        :param modal: The components you wish to show.
-        :type modal: Modal
-        """
-
-        payload: dict = {
-            "type": InteractionCallbackType.MODAL.value,
-            "data": {
-                "title": modal.title,
-                "components": modal._json.get("components"),
-                "custom_id": modal.custom_id,
-            },
-        }
-
-        await self.client.create_interaction_response(
-            token=self.token,
-            application_id=int(self.id),
-            data=payload,
-        )
-        self.responded = True
-
-    async def get_channel(self) -> Channel:
-        """
-        This gets the channel the command was invoked in.
-
-        :return: The channel as object
-        :rtype: Channel
-        """
-
-        res = await self.client.get_channel(int(self.channel_id))
-        self.channel = Channel(**res, _client=self.client)
-        return self.channel
-
-    async def get_guild(self) -> Guild:
-        """
-        This gets the guild the command was invoked in.
-
-        :return: The guild as object
-        :rtype: Guild
-        """
-
-        res = await self.client.get_guild(int(self.guild_id))
-        self.guild = Guild(**res, _client=self.client)
-        return self.guild
-
 
-class ComponentContext(CommandContext):
+class ComponentContext(ContextMixin):
```

Comment:
Personally I would make it inherit from `CommandContext`, this would have the benefit of not having the `send` and `edit` methods twice and it would still inherit from the Mixin because `CommandContext` does that.